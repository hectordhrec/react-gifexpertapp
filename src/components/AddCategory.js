import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => {
        //setInputValue(e.target.value);
        setInputValue(valor => e.target.value);
        console.log('handleInputchange');
    }

    const handleSunmit = (e) => {
        e.preventDefault();
        if (inputValue.trim().length > 2) {
            setCategories(categories => [inputValue,...categories]);
            setInputValue('');
        }
    }

    return (
        <form onSubmit={ handleSunmit }>
            <p>{inputValue}</p>
            <input type="text" className="form-control" value={inputValue} onChange={handleInputChange} />
        </form>
    );
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}

