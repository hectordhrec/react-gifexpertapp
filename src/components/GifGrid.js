import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";
import PropTypes from 'prop-types';

export const GifGrid = ({ category }) => {
    // const [images, setImages] = useState([]);

    /** Al colocar la categoría como dependencia, cada vez que la categoría cambie se va a ejecutar el useEffect */
    // useEffect(() => {
    //     getGifs(category).then(setImages);
    //     // getGifs().then(imgs => setImages(imgs)); // Otra forma
    // }, [category]);

    const { data: images, loading } = useFetchGifs(category);


    return (
        <>
            <h3 className="card animate__animated animate__fadeIn">{category}</h3>
            { loading && <p className="animate__animated animate__flash">Loading</p>}
            <div className="card-grid">
                {images.map((img) => (
                    <GifGridItem
                        key={img.id}
                        {...img}
                    />
                ))}
            </div>
        </>
    );
};

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}
