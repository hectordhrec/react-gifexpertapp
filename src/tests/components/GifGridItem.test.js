import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe('Pruebas a <GifGridItem />', () => {

    const props = {
        title : 'Ejemplo',
        url : 'https://localhost:5050'
    }

    const wrapper = shallow(<GifGridItem {...props} />);

    test('debe renderizar el componente correctamente', () => {

        expect(wrapper).toMatchSnapshot();
    });

    test('debe tener un pÃ¡rrafo con el title', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(props.title);
    });

    test('debe de tener la imagen con url y alt igual al de las props', () => {
        const img = wrapper.find('img');
        // console.log(img.props());
        // console.log(img.prop('alt'));

        expect(img.props().alt).toBe(props.title);
        expect(img.props().src).toBe(props.url);
    });

    test('debe de tener la clase animate__fadeIn', () => {
        const div = wrapper.find('div');
        // console.log(div.props());
        expect(div.prop('className').includes('animate__fadeIn')).toBe(true)
    });
    
    
    
    
});
