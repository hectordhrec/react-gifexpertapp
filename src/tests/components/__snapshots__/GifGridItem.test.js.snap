// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas a <GifGridItem /> debe renderizar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Ejemplo"
    src="https://localhost:5050"
  />
  <p>
    Ejemplo
  </p>
</div>
`;
